Allows stores to specify a cutoff time for incomplete orders

philbirt opened the pull request:
The problem being solved is really old and stale orders being returned via the API with strange state, out of stock items, among other issues.

magnusvk commented on core/app/models/concerns/spree/user_methods.rb:
```diff
@@ -23,6 +23,14 @@ module UserMethods
+      # Specifies an amount in days which incomplete orders will
+      # not be returned by #last_incomplete_spree_order based on when
+      # it was last updated. By default it will not be set, and
+      # continue to return all incomplete orders.
```
We should add `@api public` here to mark this as supported.

magnusvk commented:
I've long wanted this, so :+1: from me. I'd definitely want to use this for Bonobos.

jordan-brough commented on core/app/models/concerns/spree/user_methods.rb:
```diff
@@ -23,6 +23,15 @@ module UserMethods
+      # continue to return all incomplete orders.
+      # @api public
+      class_attribute :completable_order_cutoff
+      self.completable_order_cutoff = nil
```
I think this would be a good one for `Spree::AppConfiguration` since the concept isn't tiedt directly to the implementation of the class and it would make sense to have even if all this code was refactored.

philbirt commented:
Good point @jordan-brough, updated it to be config preferences. I also added a condition for created_at, as well as updated_at. I think it would be nice to be able to toggle on both, but interested in what others think.

jordan-brough commented:
:+1:

jhawthorn commented:
I might prefer the config name to be `_cutoff_days` just to be clear about what the integer represented.

Otherwise :+1:

I am noticing a lot of churn in this method, we should maybe consider later extracting it into a pluggable class
